ospfd/ospfd.c

```diff
@@ -630,6 +630,8 @@ static void ospf_finish_final(struct ospf *ospf)
 
 		for (ALL_LIST_ELEMENTS(red_list, node, nnode, red))
 			ospf_redistribute_unset(ospf, i, red->instance);
+
+		ospf_redist_del(ospf, i, red->instance);
 	}
```

Comment:
As you are calling ospf_redist_del() with red->instance variable, does this function call might be inside the for() loop above ?

Reply:
fixed